source/upgrade.txt

```diff
@@ -0,0 +1,101 @@
+.. _cpp-upgrade:
+
+=======================
+Upgrade Driver Versions
+=======================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+ 
+.. meta::
+   :keywords: compatibility, backwards compatibility
+
+Overview
+--------
+
+This page describes the changes you must make to your application
+when you upgrade to a new version of the {+driver-short+}.
+
+Before you upgrade, perform the following actions:
+
+- Ensure the new {+driver-short+} version is compatible with the {+mdb-server+} versions
+  your application connects to and the C++ version your
+  application runs on. For version compatibility information, see the
+  :ref:`{+driver-short+} Compatibility <cpp-compatibility-tables>`
+  page.
+- Address any breaking changes between the driver version
+  your application is using and your planned upgrade version as detailed in the
+  :ref:`Breaking Changes <cpp-breaking-changes>` section.
+
+
+.. tip::
+
+   To minimize the number of changes your application requires when
+   upgrading driver versions in the future, use the
+   :ref:`{+stable-api+} <cpp-stable-api>`.
+
+.. _cpp-breaking-changes:
+
+Breaking Changes
+----------------
+
+.. note::
+
+   The {+driver-short+} doesn't ensure API or ABI stability across patch releases
+   within a minor version.
+
+A breaking change is a change of a convention or a behavior starting in a specific
+version of the driver. This type of change may prevent your application from working
+properly if not addressed before upgrading the driver.
+
+The breaking changes in this section are categorized by the driver version that introduced
+them. When upgrading driver versions, address all the breaking changes between the current
+and upgrade versions.
+
+.. example:: Upgrading from Version 3.0
+
+  If you're upgrading the {+driver-short+} from version 3.0 to version 3.10,
+  address all breaking changes listed for versions 3.1 to 3.10, if any.
+
+.. _version-3.10-breaking-changes:
+
+Version 3.10 Breaking Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This driver version introduces the following breaking changes:
+
+- Drops support for the ``find_package(libbsoncxx)`` and ``find_package(libmongocxx)`` CMake
+  commands. Call ``find_package(bsoncxx)`` and ``find_package(mongocxx)``
+  instead.
```

Comment:
```suggestion
- Drops support for ``find_package(libbsoncxx)`` and ``find_package(libmongocxx)``. Use ``find_package(bsoncxx)`` and ``find_package(mongocxx)``
  instead.
```
Slight wording tweaks to avoid confusing the package with the command.

Reply:
We usually try to specify what an entity is, rather than just referring to it by name. What is find_package?

Reply:
[find_package()](https://cmake.org/cmake/help/latest/command/find_package.html) is a CMake command used to import an external CMake package which provides libraries, executables, and other targets as defined by the package. It is the general method for importing external dependencies (that doesn't involve build-from-source via `add_subdirectory`, a la `FetchContent`). It is comparable to `pkg-config` in purpose.

We provide CMake package config files which allow a user to `find_package()` the C++ Driver libraries into their CMake project. The package names are `bsoncxx` and `mongocxx` respectively. Those packages provide the `mongo::bsonxx_*` and `mongo::mongocxx_*` targets corresponding to the provided libraries.

In other words, we dropped support for the legacy CMake package config files (hand-written, variable-based) in favor of modern CMake package config files (generated, target-based).

Reply:
Makes sense. Thanks!